src/services/services.ts

```diff
@@ -3504,7 +3504,11 @@ namespace ts {
                         // We don't want to complete using the type acquired by the shape
                         // of the binding pattern; we are only interested in types acquired
                         // through type declaration or inference.
-                        if (rootDeclaration.initializer || rootDeclaration.type) {
+                        // Also proceed if rootDeclaration is parameter and if its containing function expression\arrow function is contextually typed -
+                        // type of parameter will flow in from the contextual type of the function
+                        if (rootDeclaration.initializer ||
+                            rootDeclaration.type ||
+                            (rootDeclaration.kind === SyntaxKind.Parameter && isExpression(rootDeclaration.parent) && typeChecker.getContextualType(<Expression>rootDeclaration.parent))) {

```

Comment:
This isn't correct in the case of object literal methods. This will also break if @sandersn's work goes in to contextually type subclass properties.

Can't you just check if you're in a parameter and you have a contextual type? Why do you need to bring the parent into this?